Library/Homebrew/os/mac/mach.rb

```diff
@@ -101,9 +99,8 @@ def mach_data
 
         mach_data << { :arch => arch, :type => type }
       end
+
       mach_data
-    rescue
-      []
```

Comment:
Good catch on that, I now leave the `rescue` as it is but remove the unnecessary `"Not a Mach-O binary."` exception.